presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushDownFilterExpressionEvaluationThroughCrossJoin.java

```diff
@@ -193,4 +217,26 @@ else if (right.containsAll(variablesInArgument1)) {
         }
         return ImmutableList.of(leftRowExpression, rightRowExpression);
     }
+
+    private List<Set<RowExpression>> getRowExpressionsFromArrayContains(RowExpression filterPredicate, List<VariableReferenceExpression> left, List<VariableReferenceExpression> right)
```

Comment:
Add row expression in `contains` function to variable reference, it will be projected below the cross join so that it can be optimized by the following CrossJoinWithArrayContainsToInnerJoin rule.